server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -403,8 +427,11 @@ public ShuffleIndexResult getShuffleIndex(
     request.setPartitionNum(partitionNum);
     request.setStorageType(storageType);
     request.setRssBaseConf(conf);
-
-    Storage storage = storageManager.selectStorage(new ShuffleDataReadEvent(appId, shuffleId, partitionId));
+    int[] range = ShuffleStorageUtils.getPartitionRange(partitionId, partitionNumPerRange, partitionNum);
+    Storage storage = storageManager.selectStorage(new ShuffleDataReadEvent(appId, shuffleId, partitionId, range[0]));
+    if (storage == null) {
+      throw new FileNotFoundException("No such data in current storage manager.");
```

Comment:
Why do we throw a `FileNotFoundException`?

Reply:
The exception of FileNotFoundException could be handled by `getLocalShuffleIndex` grpc layer, because maybe there is no local storage when directly flushing to HDFS.